cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaUpgradeDowngradeMockTest.java

```diff
@@ -211,6 +211,40 @@ public void testUpgradeWithMessageAndProtocolVersions(VertxTestContext context)
                 })));
     }
 
+    // Tests regular upgrade with the message format and protocol versions configured to the same Kafka
+    // version as we are upgrading from. IBPV and LMFV are configured to contain a "-IVx" suffix.
+    @Test
+    public void testUpgradeWithIVMessageAndProtocolVersions(VertxTestContext context)  {
+        String iv = "-IV0";
+        Kafka initialKafka = kafkaWithVersions(KafkaVersionTestUtils.PREVIOUS_KAFKA_VERSION,
+                KafkaVersionTestUtils.PREVIOUS_FORMAT_VERSION + iv,
+                KafkaVersionTestUtils.PREVIOUS_PROTOCOL_VERSION + iv);
+
+        Kafka updatedKafka = kafkaWithVersions(KafkaVersionTestUtils.LATEST_KAFKA_VERSION,
+                KafkaVersionTestUtils.PREVIOUS_FORMAT_VERSION,
+                KafkaVersionTestUtils.PREVIOUS_PROTOCOL_VERSION);
```

Comment:
Should it use the `-IV0` version here as well? Essentially, but not using it here, you limit what is tested here. The test as it is right now suggests that the user has Kafka configured with for example:
* version 2.8.0
* IBPV 2.8-IV0
* LMFV 2.8-IV0

An suddenly changes to for example:
* version 3.0.0
* IBPV 2.8
* LMFV 2.8

This seems not likely.